libbeat/tests/system/beat/beat.py

```diff
@@ -679,14 +680,46 @@ def get_elasticsearch_url(self):
 
     def get_elasticsearch_url_ssl(self):
         """
-        Returns an elasticsearch.Elasticsearch instance built from the
-        env variables like the integration tests.
+        Returns a string with the Elasticsearch URL
         """
         return "https://{host}:{port}".format(
             host=os.getenv("ES_HOST_SSL", "localhost"),
             port=os.getenv("ES_PORT_SSL", "9205"),
         )
 
+    def get_elasticsearch_template_config(self, security=True, user=None):
+        """
+        Returns a template suitable for a Beats config
+        """
+        template = {
+            "host": self.get_elasticsearch_url(),
+        }
+
+        if security:
+            template["user"] = user or os.getenv("ES_USER", "")
+            template["pass"] = os.getenv("ES_PASS", "")
+
+        return template
+
+    def get_elasticsearch_instance(self, security=True, ssl=False, url=None, user=None):
+        """
+        Returns an elasticsearch.Elasticsearch instance built from the
+        env variables like the integration tests.
+        """
+        if url is None:
+            if ssl:
+                url = self.get_elasticsearch_url_ssl()
```

Comment:
Hi @belimawr 

Thanks for mentioning this. I don't believe we need a change here because this PR preserves the original behavior. The function you're mentioning only returns a client connection to Elasticsearch. 

For configuring the actual outputter for the beat, we do that in cases where SSL is intentionally used for the test.

For example, [here is a test which does so](https://github.com/elastic/beats/blob/master/libbeat/tests/system/test_ca_pinning.py#L24-L42).

I would agree that we can probably tighten up that behavior a bit and lessen the manual configuration that would need to be done for a test author who may wish to use SSL, but I would see that as potentially being work for a follow-up PR.

WDYT?

Reply:
That makes sense.